lib/ipam/ipam.go

```diff
@@ -349,7 +281,7 @@ func (c ipams) AssignIP(args AssignIPArgs) error {
 		// Update the block using the original KVPair to do a CAS.  No need to
 		// update the Value since we have been manipulating the Value pointed to
 		// in the KVPair.
-		_, err = c.client.Backend.Update(obj)
+		_, err = c.client.Update(context.Background(), obj)
```

Comment:
also we should probably have one context in this function even if we don't expose it to the caller of `Autoassign`

Reply:
what do you mean?

Reply:
Background() returns a fixed context, so I'm not sure it makes any difference does it?